src/main/java/com/codepine/api/testrail/Request.java

```diff
@@ -114,6 +114,8 @@ public T execute() {
             String url = getUrl();
             HttpURLConnection con = (HttpURLConnection) urlConnectionFactory.getUrlConnection(url);
             con.setRequestMethod(method.name());
+            con.setFixedLengthStreamingMode(0);
```

Comment:
I assume this change will break other POST calls since the length is not 0. Your previous change was correct. Just move con.setDoOutput(true) before

`if (content != null) {`